cli/dcoscli/job/main.py

```diff
@@ -481,20 +497,22 @@ def _add_schedules(job_id, schedules_json):
     """
 
     if schedules_json is None:
-        return 1
+        raise DCOSException('Schedule JSON is required.')
 
-    for schedule in schedules_json:
-        try:
-            _post_schedule(job_id, schedule)
-        except DCOSHTTPException as e:
-            if e.response.status_code == 404:
-                emitter.publish("Job ID: '{}' does NOT exist.".format(job_id))
-            elif e.response.status_code == 409:
-                emitter.publish("Schedule already exists.")
-            else:
-                raise DCOSException(e)
-        except DCOSException as e:
+    schedule = parse_schedule_json(schedules_json)
+    try:
+        _post_schedule(job_id, schedule)
+    except DCOSHTTPException as e:
+        if e.response.status_code == 404:
+            emitter.publish("Job ID: '{}' does NOT exist.".format(job_id))
+        elif e.response.status_code == 409:
+            emitter.publish("Schedule already exists.")
+        elif e.response.status_code == 422:
```

Comment:
this will disappear when it is merged with PR835.   the main point of this PR is to look for non-array type json.   the point of PR835 is proper error handling.